### ISSUE TITLE
Normalize uid argument to set_prognoses

### ISSUE THREAD

**Issue by daniel-klein** [starsimhub/starsim#387]
`set_prognoses` is either
```python
def set_prognoses(self, sim, target_uids, source_uids=None)
```
or
```python
def set_prognoses(self, sim, uids, source_uids=None)
```

Let's pick one and be consistent.

**Comment by cliffckerr** [comment]
I don't especially like the name `set_prognoses` -- I feel this is maybe a holdover from Covasim where we pre-computed the prognoses for the entire disease course, but we won't necessarily be doing that here. In particular, what I see the `set_prognoses` function doing is:
- Appending to the infection log
- Modifying state values (e.g. `self.susceptible[uids] = False`)
- In some cases, then actually setting prognoses (e.g. `self.ti_primary[uids] = sim.ti + rr(self.pars.dur_exposed.rvs(uids) / sim.dt)`

I wonder if we should rename `make_new_cases` to `choose_cases`, and `set_prognoses` to `make_cases` or something like that?

I also don't love "cases". For infectious diseases, "case" sometimes means only a _diagnosed_ infection. I propose that we use the term `infection` in `Infection` and its subclasses, and do not define this method for `Disease`, since it may or may not be applicable. For example, I could imagine an `Obesity` module wouldn't involve a `make_new_cases` method that would flip agents from one state to another, but would instead track continuous variables, and over a certain threshold agents would be defined as being a "case".

**Comment by cliffckerr** [comment]
Closed by #527
